packages/gatsby-theme-carbon/src/components/Video/Video.js

```diff
@@ -4,32 +4,51 @@ import { settings } from 'carbon-components';
 
 const { prefix } = settings;
 
-export default class Vimeo extends React.Component {
-  static propTypes = {
-    vimeoId: PropTypes.string.isRequired,
-    title: PropTypes.string.isRequired,
-  };
+export const Video = ({ vimeoId, title, src, poster, ...props }) => {
+  const vimeoVid = (
+    <div className="gatsby-resp-iframe-wrapper">
+      <div className="embedVideo-container">
+        <iframe
+          title={title}
+          src={`https://player.vimeo.com/video/${vimeoId}`}
+          width="640"
+          height="360"
+          frameBorder="0"
+          webkitallowfullscreen="true"
+          mozallowfullscreen="true"
+          allowFullScreen
+        />
+      </div>
+    </div>
+  );
 
-  render() {
-    const { vimeoId, title } = this.props;
+  const videoVid = (
+    <video controls poster={poster} {...props}>
+      <source src={src} type="video/mp4" />
+    </video>
+  );
 
-    return (
-      <div className={`${prefix}--video`}>
-        <div className="gatsby-resp-iframe-wrapper">
-          <div className="embedVideo-container">
-            <iframe
-              title={title}
-              src={`https://player.vimeo.com/video/${vimeoId}`}
-              width="640"
-              height="360"
-              frameBorder="0"
-              webkitallowfullscreen="true"
-              mozallowfullscreen="true"
-              allowFullScreen
-            />
-          </div>
-        </div>
-      </div>
-    );
-  }
-}
+  return (
+    <div className={`${prefix}--video`}>
+      {src === '' && vimeoId === '' ? (
```

Comment:
This won't work if they're undefined.

I think we should just do this with proptypes, rather than rendering a paragraph tag maybe?